codeCainUmlEditor/src/main/Java/Relationship.java

```diff
@@ -87,78 +80,153 @@ private static boolean entryIsValid(String class1, String class2){
 
     /**
      * adds a relationship to the relationship map
+     *
      * @param class1 the first class to add
      * @param class2 the second class to add
-     * @throws Exception if a class entered doesn't exist
+     * @return true if the input is valid, false if the input is invalid.
      */
-    public static void addRelationship(String class1, String class2) throws Exception {
-
-        boolean isValid = entryIsValid(class1, class2);
+    public static boolean addRelationship(String class1, String class2){
 
-        if(!isValid) throw new Exception("invalid entry");
+        if (relationshipExists(class1,class2)){
+            System.out.println("relationship already exists");
+            return false;
+        }
+        if (!Class.classMap.containsKey(class1)){
+            System.out.println(class1 + " is not an existing class");
+            return false;
+        }
+        if (!Class.classMap.containsKey(class2)){
+            System.out.println(class2 + " is not an existing class");
+            return false;
+        }
 
         Relationship newRelationship = new Relationship(class1, class2);
 
         relationshipList.add(newRelationship);
+        System.out.println("relationship between " + class1 + " and " + class2 + " added");
+        return true;
     }
 
-
+//
```

Comment:
What are these huge chunks of commented out code for? Are they needed anymore or is it something that needs to be worked on and completed before the end of Sprint 1?

Reply:
those are just the old methods with the thrown exceptions. I left them in there just in case